### ISSUE TITLE
Error when writing null value to properties

### ISSUE THREAD

**Issue by snorfalorpagus** [conda-forge/fiona-feedstock#49]
I've come across a problem whereby attempting to set a field to NULL in a shapefile is not working. The code below triggers the bug on my machine (Windows 10, Python 3.6 64-bit, fiona 1.7.8 np112py36_0). The same code works as expected on another Linux machine where Fiona is installed without conda. This could be a bug in Fiona, GDAL or the conda-forge builds of either. Using GDAL 2.2.1 np112py36_0.

```
import fiona

meta = {"driver": "ESRI Shapefile", "schema": {"geometry": "Point", "properties": {"RETURN_P": "int"}}}
dst = fiona.open("new.shp", "w", **meta)
g = {"coordinates": [1.0, 2.0], "type": "Point"}
feature = {"geometry": g, "properties": {"RETURN_P": None}}
dst.write(feature)
dst.close()

src = fiona.open("new.shp", "r")
feature = next(src)
assert(feature["properties"]["RETURN_P"] is None) # fails
src.close()
```

**Comment by ocefpaf** [comment]
It does look like a problem with `gdal 2.2.*` do you want to report it upstream?

Meanwhile I am working on a new release of `gdal 2.1.4` so we can fix the package in `conda-forge`.

**Comment by snorfalorpagus** [comment]
The equivalent code using the osgeo.ogr module with the same build of GDAL seems to work correctly.

```
from osgeo import ogr                                 
                                                      
dr = ogr.GetDriverByName("ESRI Shapefile")            
ds = dr.CreateDataSource("test.shp")                  
layer = ds.CreateLayer("test", geom_type=ogr.wkbPoint)
                                                      
field = ogr.FieldDefn("RETURN_P", ogr.OFTInteger)     
layer.CreateField(field)                              
                                                      
VALUE = None                                          
                                                      
defn = layer.GetLayerDefn()                           
feature = ogr.Feature(defn)                           
feature.SetField("RETURN_P", VALUE)                   
layer.CreateFeature(feature)                          
feature = None                                        
layer = None                                          
ds = None                                             
                                                      
ds = dr.Open("test.shp")                              
layer = ds.GetLayer()                                 
feature = layer.GetFeature(0)                         
value = feature.GetField("RETURN_P")                  
print(value)                                          
assert(value == VALUE)
```
Does this mean it's a problem with Fiona (or the conda-forge build of Fiona)?

**Comment by ocefpaf** [comment]
> Does this mean it's a problem with Fiona (or the conda-forge build of Fiona)?

I am not sure if there are any `fiona` wheels compiled with `gdal 2.2.*`, so we cannot say if the problem is exclusive in conda-forge's `fiona`. My guess is that `fiona` is not 100% compatible with `gdal 2.2.*` yet.

**Comment by snorfalorpagus** [comment]
It looks like Fiona already has a test case that should cover this bug. https://github.com/Toblerity/Fiona/blob/master/tests/test_feature.py#L39-L44

Fiona doesn't support GDAL 2.2 yet (https://github.com/Toblerity/Fiona/pull/445) although the above test doesn't seem to be failing - the only issue is something related to GeoJSON.

**Comment by ocefpaf** [comment]
Interesting... Anyways, I am slowly changing things to be compiled against `gdal 2.1*` and, once the next version of `fiona` is out, we can test this again.

**Comment by ocefpaf** [comment]
Should be fixed in #50 (Now we need to wait for the next release of `fiona` to use `gdal 2.2.x`.)

Thanks @snorfalorpagus!
